src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -52,6 +60,8 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
             var scaffoldColumnAttribute = attributes.OfType<ScaffoldColumnAttribute>().FirstOrDefault();
             var uiHintAttribute = attributes.OfType<UIHintAttribute>().FirstOrDefault();
 
+            var localizer = _stringLocalizerFactory?.Create(context.Key.ContainerType ?? context.Key.ModelType);
```

Comment:
@ryanbrandenburg it's not _that_ expensive to create a localizer. But if you move this code within the `if`s that control whether it's needed, note `displayAttribute != null` is only the first condition. We also don't need 3 separate localizers. Suggest something like
``` c#
IStringLocalizer localizer = null;
if (displayAttribute != null)
{
    if (_stringLocalizerFactory != null &&
        displayAttribute.ResourceType == null &&
        !string.IsNullOrEmpty(displayAttribute.Name))
    {
        if (localizer == null)
        {
            var modelType = context.Key.ContainerType ?? context.Key.ModelType;
            localizer = _stringLocalizerFactory.Create(modelType);
        }

        displayMetadata.Name = () => localizer[displayAttribute.GetName()].Value;
    }
    else
    {
        displayMetadata.Name = () => displayAttribute.GetName();
    }
}
```
Of course the `localizer` setup doesn't have to be conditional in the `Description` logic. If you refactor the logic into another method, use a `ref` parameter for the localizer.